apps/openmw/mwdialogue/dialoguemanagerimp.cpp

```diff
@@ -647,6 +647,9 @@ namespace MWDialogue
             if (!info->mSound.empty())
                 sndMgr->say(actor, info->mSound);
         }
+
+        mActor = actor;
```

Comment:
It's incorrect to overwrite mActor. That is the actor that the dialogue window is opened for, but ```say``` is completely independent of that window. I suggest to add a new parameter (actor) to executeScript.